templates/pkg/crd_manager.go.tpl

```diff
@@ -168,3 +168,16 @@ func newResourceManager(
 		sdkapi:	   svcsdk.New(sess),
 	}, nil
 }
+
+// EnsureStatusDefaults ensures that the default status properties are
+// set to defaults if found nil.
+func (rm *resourceManager) EnsureStatusDefaults(
```

Comment:
If you move the responsibility of setting/adding a new Condition on the CR to the `resourceManager`, I don't believe this method will be necessary, right? Since `setStatusDefaults` will be callable by the `resourceManager` and it does all the meat of this method, which I think you only created in order to have the `reconciler` be able to call it?